subsys/task_wdt/Kconfig

```diff
@@ -5,7 +5,6 @@
 
 menuconfig TASK_WDT
 	bool "Task-level software watchdog"
-	select REBOOT
```

Comment:
This should be converted to `depends on`. Otherwise a user could enable the task watchdog without enabling `REBOOT`, leading to a no-op when no custom callback is set and the watchdog triggers. This would break previous default behavior.